cloud/scope/cluster.go

```diff
@@ -234,6 +243,11 @@ func (s *ClusterScope) RouteTable() *infrav1.RouteTable {
 	return &s.AzureCluster.Spec.NetworkSpec.GetNodeSubnet().RouteTable
 }
 
+// NatGateway returns the cluster node nat gateway.
+func (s *ClusterScope) NatGateway() *infrav1.NatGateway {
```

Comment:
If you want to return multiple nat gateways from `NatGatewaySpecs()` you can follow the example with `NodeSubnet/ControlPlaneSubnet`, or something like that.

Reply:
I want to return multiple gateways. I believe I need to add a `Role` field to the `NatGatewaySpec`, otherwise when we are reconciling NAT Gateways that have been already created and want to save the NAT Gateway ID back into the `SubnetSpec` in the CR, I wouldn't know which `NatGatewayspec` is which while iterating. Am I making sense?

Reply:
since you have access to the subnets here, I might do something like:
```
func (s* clusterScope) NodeNatGateway() *infrav1.NatGateway
func (s* clusterScope)  ControlPlaneNatGateway() *infrav1.NatGateway
func (s *ClusterScope) NatGatewaySpecs() []azure.NatGatewaySpec {
	var gateways []azure.NatGatewaySpec{}
        if gateway := s.NodeNatGateway(); gateway != nil {
            gateways = append(gateways, gateway)
        }
        if gateway := s.ControlPlaneNatGateway(); gateway != nil {
            gateways = append(gateways, gateway)
        }
}
```

Reply:
and yeah the individual funcs could attach the role to the nat gateway, cc @CecileRobertMichon on that approach